src/createServer.js

```diff
@@ -1,11 +1,233 @@
+/* eslint-disable function-paren-newline */
 'use strict';
 
-// const express = require('express');
+const express = require('express');
+const router = express.Router();
+const cors = require('cors');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+  const mockUsers = [];
+  const mockExpenses = [];
+
+  app.use(cors());
+  app.use(express.urlencoded({ extended: true }));
+  app.use(express.json());
+  app.use('/', router);
+
+  app.get('/users', (req, res) => {
+    if (mockUsers.length === 0) {
+      return res.send([]);
+    }
+    res.status(200).send(mockUsers);
+  });
+
+  app.post('/users', (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.status(400).send('Name is required!');
+
+      return;
+    }
+
+    const maxId = mockUsers.reduce(
+      (max, user) => (user.id > max ? user.id : max),
+      0,
+    );
+    const newId = maxId + 1;
+
+    const newUser = {
+      id: newId,
+      name,
+    };
+
+    mockUsers.push(newUser);
+
+    res.status(201).send(newUser);
+  });
+
+  app.get('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    const newUser = mockUsers.find((user) => user.id === Number(id));
+
+    if (!newUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+    res.status(200).send(newUser);
+  });
+
+  app.delete('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    if (!id || isNaN(Number(id))) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const userIndex = mockUsers.findIndex((user) => user.id === Number(id));
+
+    if (userIndex === -1) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    mockUsers.splice(userIndex, 1);
+    res.sendStatus(204);
+  });
+
+  app.patch('/users/:id', (req, res) => {
+    const { id } = req.params;
+    const { name } = req.body;
+
+    const newUser = mockUsers.find((user) => user.id === Number(id));
+
+    if (!newUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    if (typeof name !== 'string') {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    Object.assign(newUser, { name });
+
+    res.status(200).send(newUser);
+  });
+
+  app.get('/expenses', (req, res) => {
+    const { userId, from, to, categories } = req.query;
+
+    let filteredExpenses = mockExpenses;
+
+    if (userId) {
+      filteredExpenses = filteredExpenses.filter(
+        (expense) => expense.userId === Number(userId),
+      );
+    }
+
+    if (categories) {
+      const categoryList = categories
+        .split(',')
+        .map((category) => category.trim());
+
+      if (categoryList.length > 0) {
+        filteredExpenses = filteredExpenses.filter((expense) =>
+          categoryList.includes(expense.category),
+        );
+      }
+    }
+
+    if (from && to) {
```

Comment:
All these _if_ approaches work, but could become something hard to maintain and scale in case of more and more properties/queries needed.
You could try storing filtering functions in a hashMap (Object works), and using the given function accordingly.
This will keep all scalability in a single place and remove multiple _if_ checks. (because your code checks all ifs, even when already passing the first one).
If you do so, remember to add a default filtering function